### ISSUE TITLE
Overwrite does not raise error when we try to overwrite a non-existent node

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#938]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/938)
In GitLab by [[Gitlab user @jennis]](https://gitlab.com/jennis) on Mar 1, 2019, 11:46

Our documentation states that the (list overwrite directive)[https://docs.buildstream.build/format_intro.html#list-overwrite] `(=)`, "has the same behavior as a literal list, except that an error will be triggered in the case that there is no underlying list to overwrite". However, this does not seem to be the behaviour.

I've created commits for two separate tests which test this: 

* Test in `tests/internals/yaml.py`: https://gitlab.com/BuildStream/buildstream/commit/6b50e0356fdd694df8d9570484aab33d58ab2f12

* Test in `tests/format/include_composition.py`: https://gitlab.com/BuildStream/buildstream/commit/9a7fb3784d66b4a491f19070a94017888f1eb04b

The former test is probably clearer.

## Expected behaviour
This should raise a LoadError

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jennis]](https://gitlab.com/jennis) on Mar 1, 2019, 17:03

[[Gitlab user @tristanvb]](https://gitlab.com/tristanvb), after some investigation I am not sure that this is the case (good news I guess).

I have added a realistic test here: https://gitlab.com/BuildStream/buildstream/commit/60909d95b1abc392a94188599c425fbf0729fa42 (ignore the naming as it is actually *valid*, not invalid). This test case is something that a user should expect to work - and it does :)

To summarise: the confusion I faced was that I assumed `_yaml.compose(target, source)` would perform the composition and error if it was unable to compose the source onto the target, per `compose()` call. Especially in the case of an explicit list override. 

I think that perhaps the test I've linked in this comment could be added to the test suite but renamed to something like 'test_eventual_composition' or something similar, what are your thoughts?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jennis]](https://gitlab.com/jennis) on Mar 11, 2019, 18:00

mentioned in merge request !1221

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jennis]](https://gitlab.com/jennis) on Mar 11, 2019, 18:01

Now having understood the code, I've added some tests which test the use-case which we originally expected to fail, see !1221.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @marge-bot123]](https://gitlab.com/marge-bot123) on Mar 12, 2019, 15:13

closed via merge request !1221

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @marge-bot123]](https://gitlab.com/marge-bot123) on Mar 12, 2019, 15:13

mentioned in commit e274adf868dc538d6ae678d2f5a9e4d636475c9a
